### PR TITLE
Salesforce Marketing Cloud Connector, ServiceUri in Dynamics LS, and SSL support in MongoDB LS

### PR DESCRIPTION
**Pull request by wuchunchunt19**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json [hunk 1]
```diff
@@ -3714,6 +3726,58 @@
...
+      "required": [ 
+        "typeProperties" 
+      ] 
+    }, 
+    "SalesforceMarketingCloudLinkedServiceTypeProperties": { 
+      "description": "Salesforce Marketing Cloud linked service properties.", 
+      "properties": {
+        "clientId": {
+          "description": "The client ID associated with the Salesforce Marketing Cloud application.",
+          "type": "object"
```
Comment (alvadb) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json]: Please add the actual type of the property and do this for the other properties marked as object.

**Review by alvadb — CHANGES_REQUESTED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json (hunk 1 above)
Comment (wuchunchunt19) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json]: Sure. Thanks for the comment.

**Comment by wuchunchunt19** [comment]
I have re-pushed the changes to add Type for properties as alvadb suggested.

**Comment by alvadb** [comment]
I looked at the files and I still see "type": "object" for properties in the modified portions of the file.

**Comment by wuchunchunt19** [comment]
The "type":"object" is kind of by-design, as your can see from other existing models from these files, they all have this "type":"object" for properties, because its type could be indeterminate (String or expression or secureString). For example, in my changes, we add type information in the description (same as others):

"clientId": {
          "description": "The client ID associated with the Salesforce Marketing Cloud application. Type: string (or Expression with resultType string).",
          "type": "object"
        }

Changing the type:object for all the properties would be a non-trivial work and might break several existing models, it should be out of the scope of this PR.

**Review by alvadb — APPROVED** [review]

**Pull request merged**